src/index.js

```diff
@@ -8,3 +10,32 @@ export {
 
 export * from './service';
 export * from './message';
+
+export default (config) => {
+  const internalLogger = new Logger(config);
+
+  return (label) => {
+    const labelledLogger = internalLogger.Label(label);
+
+    const logger = level => (...args) => {
+      let msg;
+
+      if (args.length === 2) {
```

Comment:
We should not determine message depending on the arg length and should still pass in the `Message` object we need.
The connection between message and logger is through some sort of interface (though javascript doesn't really provide that).
`LogMessage` and `ErrorMessage` are meant to be extended. for instance, we can do something like
```js
class PaymentErrorMessage extends ErrorMessage {
   ...
}
```
it should override `toString()` and `toObject()` to provide context specific formatting of information.